src/allotropy/parsers/luminex_xponent/luminex_xponent_structure.py

```diff
@@ -73,26 +74,59 @@ def create(
         raw_datetime = info_row[str, "BatchStartTime"]
         sample_volume = info_row.get(str, ["SampleVolume", "MaxSampleUptakeVolume"])
 
+        # Read all the keys first
+        software_version = info_row.get(str, "Build")
+        equipment_serial_number = info_row.get(str, "SN")
+        analytical_method_identifier = info_row.get(str, "ProtocolName")
+        method_version = info_row.get(str, "ProtocolVersion")
+        experimental_data_identifier = info_row.get(str, "Batch")
+        detector_gain_setting = info_row.get(
+            str, ["ProtocolReporterGain", "ProtocolOperatingMode"]
+        )
+        data_system_instance_identifier = info_row[str, "ComputerName"]
+        analyst = info_row.get(str, "Operator")
+
+        # Mark keys that are accessed directly in helper methods
+        if "Program" in header_data:
```

Comment:
No need to put if/then statement here, you can just mark them:

```suggestion
header_data.mark_read({"Program", "ProtocolPlate"})
```